specutils/io/default_loaders/tabular_fits.py

```diff
@@ -148,7 +149,13 @@ def tabular_fits_writer(spectrum, file_name, hdu=1, update_header=False, **kwarg
 
     # Include uncertainty - units to be inferred from spectrum.flux
     if spectrum.uncertainty is not None:
-        unc = spectrum.uncertainty.quantity.to(funit, equivalencies=u.spectral_density(disp))
+        unc = (
+            spectrum
+            .uncertainty
+            .represent_as(StdDevUncertainty)
+            .quantity
+            .to(funit, equivalencies=u.spectral_density(disp))
+        )
         columns.append(unc.astype(ftype))
         colnames.append("uncertainty")
```

Comment:
```suggestion
        if spectrum.uncertainty.uncertainty_type == 'var':
            uunit = funit**2
        elif spectrum.uncertainty.uncertainty_type == 'ivar':
            uunit = funit**-2
        else:
            uunit = funit
        unc = spectrum.uncertainty.quantity.to(uunit, equivalencies=u.spectral_density(disp))
        columns.append(unc.astype(ftype))
        colnames.append(f"{spectrum.uncertainty.uncertainty_type}")
```
The original behaviour definitely needs fixing, but I'd strongly recommend to allow preserving the original type. This could be one approach to set consistent units.